docs/reference-manual/native-image/guides/optimize-file-size.md

```diff
@@ -0,0 +1,105 @@
+
+For the demo, run a "fortune teller" application that simulates the traditional `fortune` Unix program (for more information, see [fortune](https://en.wikipedia.org/wiki/Fortune_(Unix)){:target="_blank"}).
+
+1. Clone the GraalVM Demos repository:
+    ```bash
+    git clone https://github.com/graalvm/graalvm-demos.git
+    ```
+    
+2. Change directory to _fortune-demo/fortune-maven_:
+    ```bash
+    cd fortune-demo/fortune-maven
+    ```
+
+## Build a Native Executable with Default Configuration
+
+1. Create a native executable using the [Maven plugin for Native Image](https://graalvm.github.io/native-build-tools/latest/maven-plugin.html){:target="_blank"}:
+    ```bash
+    mvn -Pnative package
+    ```
+    The command compiles the project, creates a JAR file with all dependencies, and then generates a native executable, `fortune`, in the _target_ directory.
+
+2. (Optional) Run the application:
+    ```bash
+    ./target/fortune
+    ```
+    The application will return a random saying. 
+
+3. Check the file size which should be around 13M:
+    ```bash
+    du -sh target/fortune*
+    ```
+
+## Build a Native Executable Optimized for Size
+
+Next create a native executable with the size optimization on, giving a different name for the output file to differentiate it from the previous build.
+
+1. Open the _pom.xml_ file. Find the `native-maven-plugin` declaration, and add the following build arguments within the `<configuration>` element. The configuration should look like this:
+    ```xml
+    <configuration>
+        <imageName>fortune-optimized</imageName>
+        <buildArgs>
+            <buildArg>-Os</buildArg>
+            <buildArg>--emit build-report</buildArg>
+        </buildArgs>
+    </configuration>
+    ```
+    The `-Os` option enables size optimization. 
+    The option `--emit build-report` generates a build report along with other artifacts in the _target_ directory. 
+
+2. Create the second native executable:
+    ```bash
+    mvn -Pnative package
+    ```
+    The command generates an executable file, `fortune-optimized`, in the _target_ directory.
+
+3. Compare the sizes of all relevant output files:
+    ```bash
+    du -h target/fortune*
+    ```
+    You should see the output similar to this:
+    ```
+    13M    target/fortune
+    16K    target/fortune-1.0-SNAPSHOT.jar
+    9.8M   target/fortune-optimized
+    1.9M   target/fortune-optimized-build-report.html
+    ```
+    The file size decreased from 13M to 9.8M! 
+
+How much the file size can be reduced by the `-Os` option varies between applications, and depends on how much Native Image applies inlining and other optimizations that increase size in the default `-O2` mode.
+
+The build report generated in the previous step, _fortune-optimized-build-report.html_, tells exactly what was included in your native executable.
+It is an HTML file that you can open in a regular web browser. 
```

Comment:
Why not generate this report for both cases, so that readers can compare them?

Reply:
It makes sense, I agree. It'd be good to update the public demo first. I'll follow up on this suggestion in the next PR. Thanks for reviewing!